content/docs/operating/configuration.md

```diff
@@ -399,6 +403,65 @@ region: <string>
 [ port: <int> | default = 80 ]
 ```
 
+### `<openstack_sd_config>`
+
+OpenStack SD configurations allow retrieving scrape targets from OpenStack Nova
+instances.
+
+The following meta labels are available on targets during [relabeling](#relabel_config):
+
+* `__meta_openstack_instance_id`: the OpenStack instance ID
+* `__meta_openstack_instance_name`: the OpenStack instance name
+* `__meta_openstack_instance_status`: the status of the OpenStack instance
+* `__meta_openstack_instance_flavor`: the flavor of the OpenStack instance
+* `__meta_openstack_public_ip`: the public IP of the OpenStack instance
+* `__meta_openstack_private_ip`: the private IP of the OpenStack instance
+* `__meta_openstack_tag_<tagkey>`: each tag value of the instance
+
+See below for the configuration options for OpenStack discovery:
+
+```
+# The information to access the OpenStack API.
+
+# The OpenStack Region.
+region: <string>
+
+# IdentityEndpoint specifies the HTTP endpoint that is required to work with
+# the Identity API of the appropriate version. While it's ultimately needed by
+# all of the identity services, it will often be populated by a provider-level
+# function.
+[ identity_endpoint: <string> ]
+
+# Username is required if using Identity V2 API. Consult with your provider's
+# control panel to discover your account's username. In Identity V3, either
+# UserID or a combination of Username and DomainID or DomainName are needed.
+[ username: <string> ]
+[ userid: <string> ]
+
+# Password for the Identity V2 and V3 APIs. Consult with your provider's
+# control panel to discover your account's preferred method of authentication.
+[ password: <string> ]
+
+# At most one of DomainID and DomainName must be provided if using Username
+# with Identity V3. Otherwise, either are optional.
+[ domain_name: <string> ]
+[ domain_id: <string> ]
+
+# The ProjectID and ProjectName fields are optional for the Identity V2 API.
+# Some providers allow you to specify a ProjectName instead of the ProjectId.
+# Some require both. Your provider's authentication policies will determine
+# how these fields influence authentication.
+[ project_name: <string> ]
+[ project_id: <string> ]
+
+# Refresh interval to re-read the instance list.
+[ refresh_interval: <duration> | default = 60s ]
+
+# The port to scrape metrics from. If using the public IP address, this must
+# instead be specified in the relabeling rule.
```

Comment:
@cgroschupp This comment was still open, but I think I get it now: by default, the private IP + configured port are prepopulated into the `__address__` label, but if you want to use the public IP, you'll need to manually use relabeling to use the `__meta_openstack_public_ip` meta label and also specify the port in there. Fine to merge this for now I think, but could be good to explain this further at some point.